### PR TITLE
Update to Drupal 8.4

### PR DESCRIPTION
**Pull request by cjgratacos**
Adds support for Drupal 8.4 and adds local patches for asm89/stack-cors library. Reasoning behind having our version of the patch can be found [here](https://gist.github.com/cjgratacos/bfcb8638b974e44827df0c1bac333ce2).

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -49,9 +49,15 @@
...
-        "enable-patching": true
+        "enable-patching": true,
+        "patches":{
+            "asm89/stack-cors": {
+                "(November 29 2017 06:22:06 PM: See issue #223 on contentacms/contenta_jsonapi) Add origin matcher for wildcard matching": "patches/asm89_stack-cors/pull_request_42.patch"
+            }
+        }
     },
     "config": {
+        "preferred-install": "source",
```
Comment (e0ipso) [inline composer.json:60]: Why this change?

Comment (e0ipso) [reply composer.json:60]: If this is related to the git pulling of the cors library, can we do:
```
        "preferred-install": {
            "asm89/stack-cors": "source",
            "*": "auto"
        }
```

**Review by e0ipso — COMMENTED** [review]
Just one question left.

**Review by e0ipso — APPROVED** [review]

File: composer.json (hunk 1 above)
Comment (dawehner) [reply composer.json:60]: Yeah, preferred install source slows down the install process quite a bit.

Comment (e0ipso) [reply composer.json:60]: fixed via https://github.com/contentacms/contenta_jsonapi_project/commit/c7d88b10e5dc78427ac29ff1a036cf2b09ecbf70

**Comment by dawehner** [comment]
Thank you!
